packages/shared/src/logger.ts

```diff
@@ -106,6 +108,30 @@ class Logger {
     }
   }
 
+  private formatTags(tags?: Record<string, unknown>) {
+    if (!tags) {
+      return;
+    }
+
+    let result: Record<string, unknown> = {};
+
+    for (const [key, value] of Object.entries(tags)) {
+      if (value instanceof Error) {
+        result = {
+          ...result,
+          errorName: value.name,
+          errorMessage: value.message,
+          errorStack: value.stack,
+        };
```

Comment:
This would overwrite if `result` had any `errorName`, `errorMesage`. Also if there multiple error instance (unlikely) but the last one would win. What if we do something like:
```suggestion
          ...result,
          [key]: {
            ...error // if there are any extra properties.
            errorName: value.name,
            errorMessage: value.message,
            errorStack: value.stack,
          }
        };
```
This will put everything inside the same key.